tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -2715,8 +2715,82 @@ public function sendShareSpaceViaLinkRequest(
 				$body
 			)
 		);
-	
+
 		// set last response as PublicShareData. using method from core
 		$this->featureContext->setLastPublicShareData($this->featureContext->getResponseXml(null, __METHOD__));
+		// set last shareId if ShareData exists. using method from core
+		if (isset($this->featureContext->getLastPublicShareData()->data)) {
+			$this->featureContext->setLastPublicLinkShareId((string) $this->featureContext->getLastPublicShareData()->data[0]->id);
+		}
+	}
+
+	/**
+	 * @Given /^user "([^"]*)" has created a public link share of the space "([^"]*)" with settings:$/
+	 *
+	 * @param  string $user
+	 * @param  string $spaceName
+	 * @param TableNode|null $table
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 */
+	public function sendHasShareSpaceViaLinkRequest(
+		string $user,
+		string $spaceName,
+		?TableNode $table
+	): void {
+		$this->sendShareSpaceViaLinkRequest($user, $spaceName, $table);
+
+		$expectedHTTPStatus = "200";
+		$this->featureContext->theHTTPStatusCodeShouldBe(
+			$expectedHTTPStatus,
+			"Expected response status code should be $expectedHTTPStatus"
+		);
+		$this->featureContext->setLastPublicLinkShareId((string) $this->featureContext->getLastPublicShareData()->data[0]->id);
+	}
+
+	/**
+	 * @Then /^for user "([^"]*)" the space "([^"]*)" should (not|)\s?contain the last created public link$/
+	 *
+	 * @param string    $user
+	 * @param string    $spaceName
+	 * @param string    $shouldOrNot   (not|)
+	 *
+	 * @return void
+	 *
+	 * @throws Exception|GuzzleException
+	 */
+	public function userTheSpaceShouldContainLinks(
+		string $user,
+		string $spaceName,
+		string $shouldOrNot
+	): void {
+		$space = $this->getSpaceByName($user, $spaceName);
+		$url = "/apps/files_sharing/api/v1/shares";
+
+		$bodyTable = new TableNode([
+			["space_ref", $space['id']],
+			["reshares", true],
+		]);
+
+		$this->ocsContext->userSendsHTTPMethodToOcsApiEndpointWithBody(
+			$user,
+			'GET',
+			$url,
+			$bodyTable
+		);
+
+		$should = ($shouldOrNot !== "not");
+		$responseArray = json_decode(json_encode($this->featureContext->getResponseXml()->data),true, 512, JSON_THROW_ON_ERROR);
+
+		if ($should) {
+			Assert::assertNotEmpty($responseArray, __METHOD__ . ' Response should contain a link, but it is empty');
+			foreach ($responseArray as $element) {
+				$expectedLinkId = $this->featureContext->getLastPublicLinkShareId();
+				Assert::assertEquals($element["id"], $expectedLinkId, "link IDs are different");
+			}
+		} else {
+			Assert::assertEmpty($responseArray, __METHOD__ . ' Response should be empty');
+		}
```

Comment:
Note: this works for the current test scenarios that only create 1 public link in the scenario. There could be more complex test scenarios in future, that create multiple public links of different things, or with different permissions, different names etc. If that happens, then the `assertEmpty` may need to be refactored (there might be some public links, but it will need to check that the "last" created public link is not in the array. Similarly the `for` loop after `assertNotEmpty` will need to process multiple different public links, and just check that one of them matches the `expectedLinkId`

Reply:
Yes, it is. I have not scenarios with creating more that one public link. And we are using function set or get "last created public link". If we want to expand to creating more links we should collecting links to array